chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -243,11 +246,11 @@ sealed class Vec[T <: Data] private (gen: => T, val length: Int)
   def apply(idx: Int): T = self(idx)
 
   @deprecated("Use Vec.apply instead", "chisel3")
-  def read(idx: UInt)(implicit compileOptions: CompileOptions): T = apply(idx)
+  def read(idx: UInt)(implicit compileOptions: CompileOptions): T = do_apply(idx)(DeprecatedSourceInfo, compileOptions)
 
   @deprecated("Use Vec.apply instead", "chisel3")
   def write(idx: UInt, data: T)(implicit compileOptions: CompileOptions): Unit = {
-    apply(idx)(compileOptions).:=(data)(DeprecatedSourceInfo, compileOptions)
+    do_apply(idx)(DeprecatedSourceInfo, compileOptions).:=(data)(DeprecatedSourceInfo, compileOptions)
```

Comment:
Can we just not thread compile options through deprecated functions?
These should be removed eventually anyways.